virtual_rainforest/models/soil/soil_model.py

```diff
@@ -132,29 +147,122 @@ def spinup(self) -> None:
-            self.data["bulk_density"],
-            self.data["soil_moisture"],
-            self.data["soil_temperature"],
-            self.data["percent_clay"],
-            dt,
-        )
+        # Find carbon pool updates by integration
+        updated_carbon_pools = self.integrate_soil_model()
 
         # Update carbon pools (attributes and data object)
         # n.b. this also updates the data object automatically
-        self.carbon.update_soil_carbon_pools(self.data, carbon_pool_updates)
+        self.replace_soil_pools(updated_carbon_pools)
 
         # Finally increment timing
         self.next_update += self.update_interval
 
     def cleanup(self) -> None:
         """Placeholder function for soil model cleanup."""
+
+    def replace_soil_pools(self, new_pools: Dataset) -> None:
+        """Replace soil pools with previously calculated new pool values.
+
+        The state of particular soil pools will effect the rate of other processes in
+        the soil module. These processes in turn can effect the exchange rates between
+        the original soil pools. Thus, a separate update function is necessary so that
+        the new values for all soil module components can be calculated on a single
+        state, which is then updated (using this function) when all values have been
+        calculated.
+
+        Args:
+            new_pools: Array of new pool values to insert into the data object
+        """
+
+        self.data["low_molecular_weight_c"] = new_pools["new_lmwc"]
+        self.data["mineral_associated_om"] = new_pools["new_maom"]
+
+    def integrate_soil_model(self) -> Dataset:
+        """Function to integrate the soil model.
+
+        For now a single integration will be used to advance the entire soil module.
+        However, this might get split into several separate integrations in future (if
+        that is feasible).
+
+        This function unpacks the variables that are to be integrated into a single
+        numpy array suitable for integration. This order of this numpy array is as
+        follows: [low_molecular_weight_c, mineral_associated_om].
+
+        Returns:
+            A data array containing the new pool values (i.e. the values at the final
+            time point)
+
+        Raises:
+            IntegrationError: When the integration cannot be successfully completed.
+        """
+
+        # Find number of grid cells integration is being performed over
+        no_cells = self.data.grid.n_cells
+
+        # Extract update interval
+        update_time = self.update_interval.astype("timedelta64[s]").astype(float)
+        # Convert from second to day units
+        t_span = (0.0, update_time / (60.0 * 60.0 * 24.0))
+
+        # Construct vector of initial values y0
+        y0 = np.concatenate(
+            [
+                self.data["low_molecular_weight_c"].to_numpy(),
+                self.data["mineral_associated_om"].to_numpy(),
+            ]
+        )
```

Comment:
In general, you will have more than 2 pools, right? Could you define a naming convention for the pools such that they can be automatically found and concatenated, eg, all of them start with `c_pool_` or something along those lines? If you do that, you could use:

```suggestion
        y0 = np.concatenate(
            [
                self.data[name].to_numpy() 
                for name in self.data 
                if name.startswith("c_pool_")
            ]
        )
```

Naturally, `construct_full_soil_model` and functions called within will need to be adapted to use this convention. 

I understand this is a somewhat dummy/reduced soil model, but if can be designed such that it can be easily expanded with the real stuff when ready, that will save time down the line.

Reply:
Yes, the order will be as set in the `Data` object. But you can easily keep track of that order but looking at the `Data` object at any time, in particular, when reconstructing the output.